services/provider/server/server.go

```diff
@@ -590,16 +597,6 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 			storageClassData["clusterID"] = s.namespace
 			storageClassData["radosnamespace"] = cephRes.Name
 
-			nodeCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["node"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
-
-			provisionerCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["provisioner"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
```

Comment:
This check is important! even if you don't need the information again you should check the existence of the "node" and "provisioner" entries into the `cephRes` before generating names. This solves 2 problems with:

1. The assumption for the existence of specific resources on the res list. 
2. The pre generation of names based on code constants as opposed to the data-driven approach from the current code.

Reply:
We are already verifying that the Secret information exists in the `CephClient` case. I don't know why the design decision was made, but each CephClient has two entries in the resource list, once at the top level and again within each given blockpool or subvol entry. And they're always the same.